src/codeu/chat/server/Controller.java

```diff
@@ -34,9 +40,18 @@
   private final Model model;
   private final Uuid.Generator uuidGenerator;
 
-  public Controller(Uuid serverId, Model model) {
+  public PrintWriter output;
+  public static boolean writeToLog;
+
+  public Controller(Uuid serverId, Model model) throws IOException {
```

Comment:
Avoid throwing checked exceptions in constructors or public methods. Either

* handle the exception, or
* if it can only be caused by programmer error, throw an unchecked exception (a subclass of `RuntimeException`)